nifi-bootstrap/src/main/java/org/apache/nifi/bootstrap/RunNiFi.java

```diff
@@ -1027,19 +1028,24 @@ public boolean accept(final File dir, final String filename) {
         cmd.add("-Dorg.apache.nifi.bootstrap.config.log.dir=" + nifiLogDir);
         cmd.add("org.apache.nifi.NiFi");
         if (props.containsKey(NIFI_BOOTSTRAP_SENSITIVE_KEY) && !StringUtils.isBlank(props.get(NIFI_BOOTSTRAP_SENSITIVE_KEY))) {
-            cmd.add("-k " + props.get(NIFI_BOOTSTRAP_SENSITIVE_KEY));
+            File password_file = new File(confDir, "sensitive.key");
+
+            // Restrict permissions:
+            password_file.setReadable(false,false);
```

Comment:
I went with `setReadable()`/`setWritable()` because of them not tying into POSIX more than necessary. But I guess we're relying on POSIX anyway. And at either rate: checking (and erroring out on) the return values indeed is an omission.